druid/src/text/rich_text.rs

```diff
@@ -19,16 +19,17 @@ use std::sync::Arc;
 
 use super::{Attribute, AttributeSpans, TextStorage};
 use crate::piet::{
-    util, Color, FontFamily, FontStyle, FontWeight, PietTextLayoutBuilder, TextLayoutBuilder,
-    TextStorage as PietTextStorage,
+    util, Color, FontFamily, FontStyle, FontWeight, PietTextLayoutBuilder, TextLayout as _,
+    TextLayoutBuilder, TextStorage as PietTextStorage,
 };
-use crate::{ArcStr, Data, Env, FontDescriptor, KeyOrValue};
+use crate::{ArcStr, Data, Env, Event, EventCtx, FontDescriptor, KeyOrValue, TextLayout};
 
 /// Text with optional style spans.
-#[derive(Debug, Clone, Data)]
+#[derive(Clone, Data)]
 pub struct RichText {
     buffer: ArcStr,
     attrs: Arc<AttributeSpans>,
+    links: Arc<[(Range<usize>, Box<dyn Fn(&mut EventCtx, &Env)>)]>,
```

Comment:
instead of storing the text ranges I would be inclined to store the actual hit-boxes? it just feels like a small simplification.

Reply:
We don't have hit boxes when we create `RichText`. So we need to add widget_added or something, to compute the hit boxes. But still we will have to store text ranges.

Reply:
right, that make sense.

Maybe we would want to store the hit boxes in the `TextLayout` object, and compute them whenever we rebuild the layout?